datastore/protobuf/datastore.proto

```diff
@@ -127,9 +133,29 @@ message GetObsRequest {
   // TODO: add search filters for other metadata
 }
 
-
 message GetObsResponse {
   int32 status = 1;
   string error = 2; // any error description (empty on success)
   repeated Metadata2 observations = 3;
 }
+
+//---------------------------------------------------------------------------
+
+message TSMdataGroup {
+  repeated TSMetadata combos = 1;
+}
+
+message GetTSACRequest {
+  repeated string attrs = 1; // one or more TSMetadata field names
+  bool include_instances = 2; // whether to include all matching time series instances in response
+}
+
+message GetTSACResponse {
+  repeated TSMdataGroup groups = 1; // zero or more groups of unique combinations of
+  // GetTSACRequest.attrs currently existing in the storage.
+  // The combos array of each group will contain the following info depending on
+  // GetTSACRequest.include_instances:
+  // Case false (default): a single item that contains the GetTSACRequest.attrs in question, with
+  //    all other attributes set to default values (typically empty strings).
+  // Case true: all time series instances that match GetTSACRequest.attrs.
```

Comment:
I find the output quite hard to use in this case. Shouldn't there be an object at the top of the group that identifies which unique combination of parameters this group refers to? Now it is up to the use to figure this out based on the data.

Compare with how `groupBy` function generally works: it returns a map where the key identifies the group, and the value is a list of the entries. Here we just have the entries.

Reply:
But in this case the entries (in the `combos` array) include the key, so keeping it separately next to the list of entries would be redundant, wouldn't it? But the premise here is of course that the client who processes the response is always fully aware of what it asked for (in the `attrs` field in the corresponding request), and will therefore also know how to pick up that info by inspecting any item in the `combos` array, typically the first item. In the case of `include_instances=false` (the default), the `combos` array always contains exactly one item that contains only the key, all other attributes having default values, which makes sense, since you only care about the key anyway. In that case, keeping the key separately would make the redundancy even more apparent. But maybe there is a use case where it is convenient to have the key in a separate field next to `combos` after all? I.e. so you can do some processing related to the key even before starting to inspect the array?

Reply:
You have high expectations of the client... ;-)
I think always having the "key" would be helpful. An additional advantage is you could then make that part of the output the same independent of the `include_instances` flag.
In `TSMdataGroup` you could have `key` and `values` (or whatever name you prefer), where `key` is of type `TSMetadata` and `values` of type `repeated TSMetadata`. If the `include_instances` flag is false, you just leave `values` empty.
Personally, I would rename the gRPC call to `GetTSAttrGroups`

Reply:
Uhm, ok, well, I guess you have a point. I'll sleep on it :)

Reply:
Slept on it and will modify according to your suggestions 👍